src/lib/server/websearch/runWebSearch.ts

```diff
@@ -111,58 +116,69 @@ export async function runWebSearch(
 			.filter(({ link }) => !blockList.some((el) => link.includes(el))) // filter out blocklist links
 			.slice(0, MAX_N_PAGES_SCRAPE); // limit to first 10 links only
 
-		// fetch the model
-		const embeddingModel =
-			embeddingModels.find((m) => m.id === conv.embeddingModel) ?? defaultEmbeddingModel;
+		const usedSources = new Set<string>();
+		const topKClosestParagraphs = 8;
+		if (webSearch.provider.doSimilaritySearch) {
+			// fetch the model
+			const embeddingModel =
+				embeddingModels.find((m) => m.id === conv.embeddingModel) ?? defaultEmbeddingModel;
```

Comment:
I moved these two `const`s up so they'd be available to the `if` branch and the `else` branch. Otherwise, the existing code is unchanged, just moved into the `if` branch here.